src/foundry/common/utils/helpers.d.mts

```diff
@@ -126,11 +137,11 @@ export function encodeURL(path: string): string;
  * inner objects.
  *
  * @param obj - The object to expand
- * @param _d  - Track the recursion depth to prevent overflow
- *              (default: `0`)
  * @returns An expanded object
+ *
+ * @remarks contrary to the official documentation, this can also expand an array of nested `Object`s.
  */
-export function expandObject(obj: object, _d?: number): object;
+export function expandObject(obj: object): any;
```

Comment:
We generally avoid `any` except to mean "any function/any class" or so on. In this case `object` as a type actually means any object which includes arrays because arrays are objects. You can think of `object` as being the same as `Record<string, any> | any[]` if you want.

So unless `expandObject` is able to start returning something besides objects I'd turn it back into `object`. If you as a caller find it annoying that you can't safely access properties on the result of expand object you'll probably have to resort to casts or to flow checks or so on.

Reply:
I was unaware arrays were object. I'll change it back.